meiga/result.py

```diff
@@ -75,43 +86,42 @@ def is_success(self) -> bool:
     def is_failure(self) -> bool:
         return not self._is_success
 
-    def throw(self):
+    def throw(self) -> Union[NoReturn, None]:
         if not self._is_success:
             raise self.value
+        return None
 
-    def unwrap(self):
+    def unwrap(self) -> Union[TS, None]:
         if not self._is_success:
             return None
-        else:
-            return self.value
+        return cast(TS, self.value)
 
-    def unwrap_or(self, failure_value: Any):
+    def unwrap_or(self, failure_value: Any) -> Union[TS, Any]:
```

Comment:
I think we can use a TypeVar to better typify the failure_value with the same type as the return value.

If we define something like the following:
 ```python
TEF = TypeVar("TEF")
```
And then use `def unwrap_or(self, failure_value: TEF) -> Union[TS, TEF]:` instead of `def unwrap_or(self, failure_value: Any) -> Union[TS, Any]`